test/Models/Catalog/esri/ArcGisMapServerCatalogItemSpec.ts

```diff
@@ -467,6 +468,115 @@ describe("ArcGisMapServerCatalogItem", function () {
       expect(item.startTime).toBe("2004-11-26T09:43:22.000000000Z");
       expect(item.stopTime).toBe("2019-11-03T14:00:00.000000000Z");
     });
+
+    it("can load a time-enabled layer without interval specified by model dimensions", async function () {
+      runInAction(() => {
+        item = new ArcGisMapServerCatalogItem("test", new Terria());
+        item.setTrait(
+          CommonStrata.definition,
+          "url",
+          "http://example.com/cadastre_history/MapServer"
+        );
+      });
+
+      const modelJson = {
+        modelDimensions: [
+          {
+            id: "Select time window duration",
+            name: "Select time window duration",
+            options: [
+              {
+                id: "no time window duration",
+                value: {}
+              }
+            ],
+            selectedId: "no time window duration"
+          }
+        ]
+      };
+      updateModelFromJson(item, CommonStrata.definition, modelJson);
+      await item.loadMapItems();
+      const expectedTimeQueryString = 1572789600000; // from json file
+      const imageryProvider = item.mapItems[0]
+        .imageryProvider as ArcGisMapServerImageryProvider;
+      expect(imageryProvider.parameters.time).toBe(expectedTimeQueryString);
+    });
+
+    it("can load a time-enabled layer with current time and forward interval specified by model dimensions", async function () {
+      runInAction(() => {
+        item = new ArcGisMapServerCatalogItem("test", new Terria());
+        item.setTrait(
+          CommonStrata.definition,
+          "url",
+          "http://example.com/cadastre_history/MapServer"
+        );
+      });
+      const modelJson = {
+        modelDimensions: [
+          {
+            id: "Select a forward time window duration",
+            name: "Select a forward time window duration",
+            options: [
+              {
+                id: "2 weeks",
+                value: {
+                  timeWindowDuration: 2,
+                  timeUnit: "week"
+                }
+              }
+            ],
+            selectedId: "2 weeks"
+          }
+        ]
+      };
```

Comment:
You don't need to test `modelDimensions` specifically. Instead you can change the `modelJson` to simply `{"timeDurationWindow": "..."}` or simply do `item.setTrait(CommonStrata.user, "timeWindowDuration", "...")`. Same for the other specs.

Reply:
It appears that if an item has `selectedId` set in `modelDimensions`  in the catalog json, the item's traits `timeWindowDuration` and `timeUnit` are still undefined after being added to the workbench. The user has to explicitly select an option from the drop-down menu to update the model. Another problem is that if an is selected, it can not be unselected by choosing an empty option value.

Reply:
You are right. That is because the `modelDimensions` is meant as an easy way to generate a UI for updating traits. So it'll change the trait value only when a user interacts with it. (Without `modelDimensions` you'll have to write a new React component to just for every custom configuration)

If you want to configure a default time window configuration, you can do like you normally configure any trait value - by specifying it in the item catalog definition. eg:
```json
{
   "type": "esri-mapServer",
   "url": "...",
   "timeWindowDuration": "4d"
}
```

> Another problem is that if an option is selected, it can not be unselected by choosing an empty option value. E.g. the following option will do nothing
You should be able to use `null` values to unset a trait.
eg: ```json
{
   "id": "no time window",
    "value": { "timeWindowDuration": null }
}
```

Reply:
It can not be `null` because its type is `number` or `undefined`. I have to do this:
```
{
  "id": "no time frame",
  "value": {
    "timeWindowDuration": 0,
    "timeUnit": "year"
  }
}
```

Reply:
You could try defining the trait with `isNullable: true` to allow `null` values:

```js
@primitiveTrait({
    name: "Time Window Duration",
    description:
      "Specify a time window duration from the current time that this dataset is available. Positive: forward; negative: backward.",
    type: "number",
    isNullable: true,
  })
  timeWindowDuration?: number;
```

Reply:
This approach seems not working as it will not try to update the model if the trait is null.